File: overriding.cpp
#include<iostream>
using namespace std;

class Base
{
    public:
        virtual void fun()
        {
            cout<<"Inside base fun\n";
        }
        void gun()
        {
            cout<<"Inside base gun";
        }

};
class Derived : public Base{

    public:
            void fun()
            {
                cout<<"Derived fun";
            }
};

int main()
{
    Base bobj;
    Derived dobj;

    Base *bp = new Derived();
    bp->fun();
    return 0;
}

File: overriding.java
class Demo
{
    public void fun()
    {
        System.out.println("In fun method of base class");
    }
    public void gun()
    {
        System.out.println("In gun method od base class ");
    }
}
class Derived extends Demo{

    public void fun()
    {
        System.out.println("In fun method of Derived class");
    }
    
}
class overriding
{

    public static void main(String arg[]) {
        Demo bobj = new  Derived();
        bobj.fun();      
    }
}